privval/signer_requestHandler.go

```diff
@@ -22,20 +22,12 @@ func DefaultValidationRequestHandler(
 	switch r := req.Sum.(type) {
 	case *pvproto.Message_PubKeyRequest:
 		if r.PubKeyRequest.GetChainId() != chainID {
-			res = mustWrapMsg(&pvproto.PubKeyResponse{
-				PubKeyType: "", PubKeyBytes: []byte{}, Error: &pvproto.RemoteSignerError{
-					Code: 0, Description: "unable to provide pubkey type and bytes",
-				},
-			})
-			return res, fmt.Errorf("want chainID: %s, got chainID: %s", r.PubKeyRequest.GetChainId(), chainID)
+			return chainIDMismatchError(r.PubKeyRequest.GetChainId(), chainID)
 		}
 
 		var pubKey crypto.PubKey
-		pubKey, err = privVal.GetPubKey()
-		if err != nil {
```

Comment:
My assumption is that this `if` condition was introduced by mistake. The correct response would be to return a wrapped error below.